### PR TITLE
Option to create lineup-upload csv file.

### PR DESCRIPTION
**Pull request by bissont**
Requires one to download the template, which has player ids, which serves as
input with --pids option. Output is the list of lineups created written to a
 a file data/current-upload.csv file, which can be uploaded as a lineup.

Usage:
Download upload template from draft-kings
//Run optimizer with new option to:
python optimize.py -w 2 --pids ~/Desktop/DKSalaries.csv  -mp 100
//upload file "data/current-upload.csv to draftkings to generate the lineups created by this tool.

### CODE REVIEW

File: optimize.py [hunk 1]
```diff
@@ -195,13 +198,74 @@ def _check_missing_players(all_players, min_cost, e_raise):
...
+    with open(pid_file, "r") as f:
+        n = 0
+        for line in f.readlines():
+            n += 1
+            if 'TeamAbbrev' in line: # line with field names was found
+                fields= line.split(',')
+                break
+        f.close()
+        f = islice(open(pid_file, "r"), n, None)
+        reader = csv.DictReader(f, fieldnames=fields)
```
Comment (BenBrostoff) [inline optimize.py]: This errors out if you don't hit the conditional in 214.

File: optimize.py [hunk 2]
```diff
@@ -195,13 +198,74 @@ def _check_missing_players(all_players, min_cost, e_raise):
...
+        player_ids = []
+        flexid=-1
+        rbs=0
+        wrs=0
+
+        for p in players:
+            # For the players in the roster, get their id.
+            # they are sorted when they are printed out, so the order is good, excepet
+            # for the flex player
+            id = player_map[p.name + " " + p.pos]
```
Comment (BenBrostoff) [inline optimize.py]: Minor, but I always try to avoid overwriting `id`since it's a built in function in Python.

File: optimize.py (hunk 1 above)
Comment (BenBrostoff) [inline optimize.py]: In general, we need more in the README or some type of other instruction about what you need to download to run this. DraftKings calls both the normal CSV download DKSalaries.csv and the template option from https://www.draftkings.com/lineup/upload the same thing, even though these CSVs are structured totally differently.

File: optimize.py [hunk 3]
```diff
@@ -195,13 +198,74 @@ def _check_missing_players(all_players, min_cost, e_raise):
                                                 str(total_report))
         raise Exception('Total missing players at price point: ' + str(miss))
 
+def create_upload_file_and_map_pids(pid_file, test_mode=False):
+    # create upload file
+    csv_name = 'test' if test_mode else 'current'
```
Comment (BenBrostoff) [inline optimize.py]: I kept the test mode for writing unit tests, so no need for it if you don't plan to unit test this.

File: optimize.py (hunk 1 above)
Comment (BenBrostoff) [inline optimize.py]: Basically, to run this feature you need to do a few things:

1) Download the "normal" CSV available this repo has always been using
2) Run `new_week.sh`
3) Download the "special" CSV, get the file location and run `python optimize.py -w 2 -pids <special_location>`.

**Comment by BenBrostoff** [comment]
First off, a huge thanks for building this feature. I used it a bunch tonight in building my lineups for Sunday. I'm going to merge in (feel free to look at my comments if you want, they were mainly just notes for myself). This is great and contribute anytime here.

**Comment by bissont** [comment]
Thanks for the code review and comments. I'll clean up the code based on your comments and also  update the readme. 

This is a really cool project you created. Thanks!

**Pull request merged**